Cython/Compiler/PyrexTypes.py

```diff
@@ -4221,7 +4221,49 @@ def cpp_optional_check_for_null_code(self, cname):
         return "(%s.has_value())" % cname
 
 
-class CppScopedEnumType(CType):
+class EnumMixin(object):
+    """
+    Common implementation details for C and C++ enums.
+    """
+
+    def create_to_py_utility_code_implementation(self, env):
```

Comment:
The semantic difference between `create_to_py_utility_code` and `create_to_py_utility_code_implementation` isn't exactly obvious. I propose making it specific to the `EnumMixin` by renaming it to:
```suggestion
    def create_enum_to_py_utility_code(self, env):
```